low-liquidity-market-attack-monitor/src/agent.js

```diff
@@ -0,0 +1,204 @@
+
+async function getCompoundTokens(
+  provider,
+  comptrollerContract,
+  compTokenAbi,
+  excludeAddresses,
+  compTokens,
+) {
+  let compTokenAddresses = await comptrollerContract.getAllMarkets();
+  compTokenAddresses = compTokenAddresses
+    .map((addr) => addr.toLowerCase())
+    .filter((addr) => excludeAddresses.indexOf(addr) === -1)
+    .filter((addr) => !Object.keys(compTokens).includes(addr));
+
+  const promises = compTokenAddresses.map(async (tokenAddress) => {
+    const contract = new ethers.Contract(tokenAddress, compTokenAbi, provider);
+    const symbol = await contract.symbol();
+    const underlying = await contract.underlying();
+
+    // eslint-disable-next-line no-param-reassign
+    compTokens[tokenAddress] = {
+      symbol,
+      underlying,
+    };
+  });
+  await Promise.all(promises);
+}
+
+function provideInitialize(data) {
+  return async function initialize() {
+    /* eslint-disable no-param-reassign */
+    // assign configurable fields
+    data.protocolName = config.protocolName;
+    data.protocolAbbreviation = config.protocolAbbreviation;
+    data.developerAbbreviation = config.developerAbbreviation;
+
+    const { excludeAddresses } = config;
+    data.excludeAddresses = excludeAddresses.map((addr) => addr.toLowerCase());
+
+    data.provider = getEthersBatchProvider();
+
+    const {
+      Comptroller: comptroller,
+      CompoundToken: compToken,
+    } = config.contracts;
+
+    // create an ethers.js Contract Object to interact with the Comptroller contract
+    const comptrollerAbi = getAbi(comptroller.abiFile);
+    data.comptrollerContract = new ethers.Contract(
+      comptroller.address,
+      comptrollerAbi,
+      data.provider,
+    );
+
+    data.compTokenAbi = getAbi(compToken.abiFile);
+    data.compTokens = {};
+
+    // from the Comptroller contract, get all of the cTokens
+    await getCompoundTokens(
+      data.provider,
+      data.comptrollerContract,
+      data.compTokenAbi,
+      data.excludeAddresses,
+      data.compTokens,
+    );
+  };
+}
+
+function provideHandleTransaction(data) {
+  return async function handleTransaction(txEvent) {
+    const {
+      protocolName,
+      protocolAbbreviation,
+      developerAbbreviation,
+      provider,
+      excludeAddresses,
+      compTokens,
+      compTokenAbi,
+      comptrollerContract,
+    } = data;
+    const findings = [];
+
+    await getCompoundTokens(
+      provider,
+      comptrollerContract,
+      compTokenAbi,
+      excludeAddresses,
+      compTokens,
+    );
+
+    const transferEvents = filterLog(txEvent.logs, ERC20_TRANSFER_EVENT);
+
+    transferEvents.forEach((transferEvent) => {
+      Object.entries(compTokens).forEach(([compTokenAddress, compToken]) => {
+        if (transferEvent.address === compToken.underlying
+          && transferEvent.args.to === compTokenAddress) {
+          filterLog(txEvent.logs, CERC20_MINT_EVENT)
+            .filter((mintEvent) => mintEvent.address === compTokenAddress)
```

Comment:
The `filterLog` function in this module has the ability to filter based on log address, so these two lines can probably be combined:
```suggestion
          filterLog(txEvent.logs, CERC20_MINT_EVENT, compTokenAddress)
```